### PR TITLE
Merge with master to pick up bugfix

### PR DESCRIPTION
**Pull request by jean-philippe-martin**
By "merge with master" I meant "merge changes _from_ master". This will still live in gcs-nio for now, though clearly we're feeling the pain from not having merged yet.

### CODE REVIEW

**Comment by jean-philippe-martin** [comment]
Ah I need to update the package names, just a moment.

**Comment by jean-philippe-martin** [comment]
@googlebot, you're broken, this is far from my first contribution to a Google project.

**Comment by mziccard** [comment]
@jean-philippe-martin you should also update versions to match the parent pom. The bot is complaining about the fact that you imported commits that are not yours, just ignore it as they all come from people that signed the CLA.

File: gcloud-java-core/src/main/java/com/google/cloud/AuthCredentials.java [hunk 1]
```diff
@@ -59,6 +76,13 @@
...
           this.getAccessToken = tokenResultClass.getMethod("getAccessToken");
+          this.getExpirationTime = tokenResultClass.getMethod("getExpirationTime");
+          this.account = (String) serviceClass.getMethod("getServiceAccountName")
+              .invoke(appIdentityService);
+          this.signForApp = serviceClass.getMethod("signForApp", byte[].class);
+          Class<?> signingResultClass = Class.forName(
+              "com.google.appengine.api.appidentity.AppIdentityService$SigningResult");
+          this.getSignature = signingResultClass.getMethod("getSignature");
           this.scopes = null;
         } catch (Exception e) {
```
Comment (mziccard) [inline gcloud-java-core/src/main/java/com/google/cloud/AuthCredentials.java]: Codacy Issue found: [Avoid throwing raw exception types.](https://www.codacy.com/app/mziccard/gcloud-java/file/2423054492/issues/source?bid=3123107&fileBranchId=3239177#l88)

File: gcloud-java-contrib/gcloud-java-nio/src/main/java/com/google/cloud/storage/contrib/nio/CloudStorageFileSystemProvider.java [hunk 2]
```diff
@@ -237,7 +232,7 @@ private SeekableByteChannel newReadChannel(Path path, Set<? extends OpenOption>
```
Comment (mziccard) [inline gcloud-java-contrib/gcloud-java-nio/src/main/java/com/google/cloud/storage/contrib/nio/CloudStorageFileSystemProvider.java]: Codacy Issue found: [The method newWriteChannel() has an NPath complexity of 336](https://www.codacy.com/app/mziccard/gcloud-java/file/2423054461/issues/source?bid=3123107&fileBranchId=3239177#l232)

**Comment by jean-philippe-martin** [comment]
Oh, it's awesome how Travis found a problem with the merged comments! Fixed now. Hopefully that's the last fix we needed.

**Comment by mziccard** [comment]
@jean-philippe-martin Looks fine, could you please squash the commits you added after the merge into only one? (from 75a22c9 to 03afeb8 inclusive)

**Comment by jean-philippe-martin** [comment]
Squished! Merging as soon as the tests are green.

**Pull request merged**